File: ruby/src/sum-of-digits-of-string-after-convert.rb
# frozen_string_literal: true

# @param {String} s
# @param {Integer} k
# @return {Integer}
def get_lucky(s, k) = k.times.reduce(convert(s)) { transform(_1) }

def convert(s) = s.chars.map { _1.ord - 'a'.ord + 1 }.join.to_i

def transform(n) = n.digits.sum

get_lucky("iiii", 1) => 36
get_lucky("leetcode", 2) => 6
get_lucky("zbax", 2) => 8

File: dart/sum-of-digits-of-string-after-convert.dart
class Solution {
  int getLucky(String s, int k) {
    return int.parse(transform(convert(s), k));
  }

  String convert(String s) {
    return s.runes.map((int c) => c - 96).join("");
  }

  String transform(String s, int k) {
    return Iterable.generate(k).fold<String>(s, (String acc, _) {
      return acc.runes.fold<int>(0, (int acc, int char) {
        var code = char - 48;
        return acc + (code ~/ 10) + (code % 10);
      }).toString();
    });
  }
}
